langchain/src/schema/runnable/remote.ts

```diff
@@ -0,0 +1,312 @@
+        metadata: obj.metadata,
+      });
+
+    if (isSuperset(keys, new Set(["content", "type", "is_chunk"]))) {
+      if (!obj.is_chunk) {
+        if (obj.type === "human") {
+          return new HumanMessage({
+            content: obj.content,
+          });
+        }
+        if (obj.type === "system") {
+          return new SystemMessage({
+            content: obj.content,
+          });
+        }
+        if (obj.type === "chat") {
+          return new ChatMessage({
+            content: obj.content,
+            role: obj.role,
+          });
+        }
+        if (obj.type === "function") {
+          return new FunctionMessage({
+            content: obj.content,
+            name: obj.name,
+          });
+        }
+        if (obj.type === "ai") {
+          return new AIMessage({
+            content: obj.content,
+          });
+        }
+      } else {
+        if (obj.type === "human") {
+          return new HumanMessageChunk({
+            content: obj.content,
+          });
+        }
+        if (obj.type === "system") {
+          return new SystemMessageChunk({
+            content: obj.content,
+          });
+        }
+        if (obj.type === "chat") {
+          return new ChatMessageChunk({
+            content: obj.content,
+            role: obj.role,
+          });
+        }
+        if (obj.type === "function") {
+          return new FunctionMessageChunk({
+            content: obj.content,
+            name: obj.name,
+          });
+        }
+        if (obj.type === "ai") {
+          return new AIMessageChunk({
+            content: obj.content,
+          });
+        }
+      }
+    }
+    if (isSuperset(keys, new Set(["text"]))) {
+      return new StringPromptValue(obj.text);
+    }
+    if (isSuperset(keys, new Set(["messages"]))) {
+      return new ChatPromptValue({
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        messages: obj.messages.map((msg: any) => revive(msg)),
+      });
+    }
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const innerRevive: (key: string) => [string, any] = (key: string) => [
+      key,
+      revive(obj[key]),
+    ];
+    const rtn = Object.fromEntries(keysArr.map(innerRevive));
+    return rtn;
+  }
+  return obj;
+}
+function deserialize<RunOutput>(str: string): RunOutput {
+  const obj = JSON.parse(str);
+  return revive(obj);
+}
+
+function withoutCallbacks(
+  options?: RunnableConfig
+): Omit<RunnableConfig, "callbacks"> {
+  const rest = { ...options };
+  delete rest.callbacks;
+  return rest;
+}
+
+export class RemoteRunnable<
+  RunInput,
+  RunOutput,
+  CallOptions extends RunnableConfig
+> extends Runnable<RunInput, RunOutput, CallOptions> {
+  private url: string;
+
+  private options?: RemoteRunnableOptions;
+
+  lc_namespace = ["langchain", "schema", "runnable", "remote"];
+
+  constructor(url: string, options?: RemoteRunnableOptions) {
+    super();
```

Comment:
Pass `fields` here for serialization - I would actually define the signature as:

```typescript
  constructor(fields: {url: string, options: RemoteRunnableOptions}) {
    super(fields);
```